apps/web/src/hooks/useOptInRedirect.ts

```diff
@@ -1,31 +1,52 @@
 import { useCallback, useLayoutEffect } from 'react';
 import { useLocation } from 'react-router-dom';
-import { FeatureFlagsKeysEnum, NewDashboardOptInStatusEnum } from '@novu/shared';
+import { NewDashboardOptInStatusEnum } from '@novu/shared';
 
 import { useNewDashboardOptIn } from './useNewDashboardOptIn';
 import { ROUTES } from '../constants/routes';
-import { useFeatureFlag } from './useFeatureFlag';
-import { IS_EE_AUTH_ENABLED } from '../config';
-
-const ROUTES_THAT_REDIRECT_TO_DASHBOARD = [ROUTES.WORKFLOWS];
+import { IS_EE_AUTH_ENABLED, NEW_DASHBOARD_URL } from '../config';
+import { useEnvironment } from './useEnvironment';
 
 export const useOptInRedirect = () => {
   const { pathname } = useLocation();
-  const { status, isLoaded, redirectToNewDashboard } = useNewDashboardOptIn();
+  const { environment } = useEnvironment();
+  const { status, isLoaded } = useNewDashboardOptIn();
+
+  const getNewDashboardUrl = useCallback(
+    (currentRoute: string): string | undefined => {
+      const newDashboardUrl = NEW_DASHBOARD_URL || window.location.origin;
+
+      switch (currentRoute) {
+        case ROUTES.GET_STARTED:
+          return `${newDashboardUrl}/env/${environment?.slug}/welcome`;
+        case ROUTES.WORKFLOWS:
+          return `${newDashboardUrl}/env/${environment?.slug}/workflows`;
+        case ROUTES.ACTIVITIES:
+          return `${newDashboardUrl}/env/${environment?.slug}/activity-feed`;
+        case ROUTES.INTEGRATIONS:
+          return `${newDashboardUrl}/integrations`;
+        case ROUTES.API_KEYS:
+          return `${newDashboardUrl}/env/${environment?.slug}/api-keys`;
+        default:
+          return undefined;
```

Comment:
Based on the Web application's current route calculate the new Dashboard URL and redirect the user when he is opt-in.

Reply:
I had to refactor the previous code because we want to allow editing the v1 workflows in the old Web app if the user is opt-in. Also at the same time clicking in the side navigation in the Web app should redirect user to the new Dashboard corresponding page.